admin/resources/js/components/poolCandidate/CreatePoolCandidate.tsx

```diff
@@ -0,0 +1,338 @@
+import { isEmpty } from "lodash";
+import {
+  CreatePoolCandidateInput,
+  Pool,
+  LanguageAbility,
+  WorkRegion,
+  SalaryRange,
+  PoolCandidateStatus,
+  User,
+  OperationalRequirement,
+  CmoAsset,
+  Classification,
+  useCreatePoolCandidateMutation,
+  CreatePoolCandidateMutation,
+  useGetClassificationsQuery,
+  useGetCmoAssetsQuery,
+  useGetOperationalRequirementsQuery,
+  useAllUsersQuery,
+  useGetPoolsQuery,
+  Maybe,
+  PoolCandidate,
+} from "../../api/generated";
+import errorMessages from "../form/errorMessages";
+import Submit from "../form/Submit";
+import Select from "../form/Select";
+import Input from "../form/Input";
+import { enumToOptions, notEmpty } from "../../helpers/util";
+import MultiSelect from "../form/MultiSelect";
+
+type Option<V> = { value: V; label: string };
+
+type FormValues = Pick<
+  PoolCandidate,
+  | "cmoIdentifier"
+  | "expiryDate"
+  | "hasDiploma"
+  | "hasDisability"
+  | "isIndigenous"
+  | "isVisibleMinority"
+  | "isWoman"
+  | "languageAbility"
+  | "expectedSalary"
+  | "locationPreferences"
+  | "status"
+> & {
+  acceptedOperationalRequirements: string[] | undefined;
+  cmoAssets: string[] | undefined;
+  expectedClassifications: string[] | undefined;
+  pool: string;
+  user: string;
+};
+interface CreatePoolCandidateFormProps {
+  classifications: Classification[];
+  cmoAssets: CmoAsset[];
+  locale: "en" | "fr";
+  operationalRequirements: OperationalRequirement[];
+  pools: Pool[];
+  users: User[];
+  handleCreatePoolCandidate: (
+    data: CreatePoolCandidateInput,
+  ) => Promise<CreatePoolCandidateMutation["createPoolCandidate"]>;
+}
+
+export const CreatePoolCandidateForm: React.FunctionComponent<CreatePoolCandidateFormProps> =
+  ({
+    classifications,
+    cmoAssets,
+    locale,
+    operationalRequirements,
+    pools,
+    users,
+    handleCreatePoolCandidate,
+  }) => {
+    const methods = useForm<FormValues>();
+    const { handleSubmit } = methods;
+
+    const formValuesToSubmitData = (
+      values: FormValues,
+    ): CreatePoolCandidateInput => ({
+      ...values,
+      acceptedOperationalRequirements: {
+        sync: values.acceptedOperationalRequirements,
+      },
+      cmoAssets: {
+        sync: values.cmoAssets,
+      },
+      expectedClassifications: {
+        sync: values.expectedClassifications,
+      },
+      pool: { connect: values.pool },
+      user: { connect: values.user },
+    });
+
+    const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+      await handleCreatePoolCandidate(formValuesToSubmitData(data))
+        .then(() => {
+          // TODO: Navigate to pool candidate dashboard.
+        })
+        .catch(() => {
+          // Something went wrong with handleCreatePoolCandidate.
+          // Do nothing.
+        });
+    };
+
+    const cmoAssetOptions: Option<string>[] = cmoAssets.map(({ id, key }) => ({
+      value: id,
+      label: key,
```

Comment:
Should probably use name[locale] over key, here. (Same goes for UpdatePoolCandidate).